app/controllers/TrusteesNinoController.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.TrusteesNinoFormProvider
+import javax.inject.Inject
+import models.Mode
+import navigation.Navigator
+import pages.{Trustees, TrusteesNamePage, TrusteesNinoPage}
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.TrusteesNinoView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TrusteesNinoController @Inject()(
+                                        override val messagesApi: MessagesApi,
+                                        sessionRepository: SessionRepository,
+                                        navigator: Navigator,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        validateIndex : IndexActionFilterProvider,
+                                        requiredAnswer: RequiredAnswerActionProvider,
+                                        formProvider: TrusteesNinoFormProvider,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        view: TrusteesNinoView
+                                    )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  val form = formProvider()
+
+  private def routes(index : Int) =
+      identify andThen
+      getData andThen
+      requireData andThen
+      validateIndex(index, Trustees) andThen
+      requiredAnswer(RequiredAnswer(TrusteesNamePage(index)))
```

Comment:
@jassalrichy Can you make `RequiredAnswer()` accept the second argument to send them to `routes.TrusteeNameController.onPageLoad()` instead of session expired. This will then daisy chain up to the correct place for that trustee.

Reply:
Done